src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -34,6 +34,26 @@ public class PooledQueueCache<TQueueMessage, TCachedMessage>
         private readonly ICacheDataAdapter<TQueueMessage, TCachedMessage> cacheDataAdapter;
         private readonly ICacheDataComparer<TCachedMessage> comparer;
 
+        public TCachedMessage? Newest
+        {
+            get
+            {
+                if (IsEmpty)
+                    return null;
+                return messageBlocks.First.Value.NewestMessage;
```

Comment:
>this is probably not multi-threaded

Yes, cache expects single threaded access.

>without the need to check for empty

Unfortunately the IsEmpty is a bit more involved than just a null check.

Good to know we can use c6 now though.